mysql-test/main/func_json.result

```diff
@@ -5263,4 +5263,205 @@ SET @obj1='{ "a": 1,"b": 2,"c": 3}';
+'123' IS JSON
+1
+SELECT 'null' IS JSON;
+'null' IS JSON
+1
+SELECT 'invalid' IS JSON;
+'invalid' IS JSON
+0
+SELECT '{"key1":1, "key2":[2,3]' IS JSON;
+'{"key1":1, "key2":[2,3]' IS JSON
+0
+SELECT '[1, 2' IS JSON;
+'[1, 2' IS JSON
+0
+SELECT 'NULL' IS JSON;
+'NULL' IS JSON
+0
+SELECT '"a",1,1,2]' IS JSON WITH UNIQUE KEYS;
+'"a",1,1,2]' IS JSON WITH UNIQUE KEYS
+0
+SELECT 'invalid' IS NOT JSON;
+'invalid' IS NOT JSON
+1
+SELECT '{"key1":1, "key2":[2,3]' IS NOT JSON;
+'{"key1":1, "key2":[2,3]' IS NOT JSON
+1
+SELECT '[1, 2]' IS NOT JSON;
+'[1, 2]' IS NOT JSON
+0
+SELECT '{"key1":1, "key2":[2,3]}' IS NOT JSON;
+'{"key1":1, "key2":[2,3]}' IS NOT JSON
+0
+SELECT '"a",1,1,2]' IS NOT JSON WITH UNIQUE KEYS;
+'"a",1,1,2]' IS NOT JSON WITH UNIQUE KEYS
+1
+# Type constraints
+SELECT js,
+js IS JSON "json?",
+js IS JSON VALUE "value?",
+js IS JSON SCALAR "scalar?",
+js IS JSON OBJECT "object?",
+js IS JSON ARRAY "array?"
+FROM (VALUES
+('123'), ('"string"'), ('{"key":1}'), ('[1,2,3]'), ('[]'), ('{}'),
+('true'), ('false'), ('null')) foo(js);
+js	json?	value?	scalar?	object?	array?
+123	1	1	1	0	0
+"string"	1	1	1	0	0
+{"key":1}	1	1	0	1	0
+[1,2,3]	1	1	0	0	1
+[]	1	1	0	0	1
+{}	1	1	0	1	0
+true	1	1	1	0	0
+false	1	1	1	0	0
+null	1	1	1	0	0
+# UNIQUE KEYS constraint
+SELECT js,
+js IS JSON OBJECT WITH UNIQUE KEYS "object_with_unique",
+js IS JSON OBJECT WITHOUT UNIQUE KEYS "object_without_unique",
+js IS JSON ARRAY WITH UNIQUE KEYS "array_with_unique",
+js IS JSON ARRAY WITHOUT UNIQUE KEYS "array_without_unique"
+FROM (VALUES
+('{"a":1, "b":2, "c":3}'),
+('{"a":1, "a":2, "a":3}'),
+('[{"a":"1"},{"c":"2","c":"3"}]'),
+('[{"a":"1"},{"b":"2","c":"3"}]'),
+('[1,2,3]'), ('[1,1,1]'), ('[]'), ('123'),
+('"string"'), ('true'), ('null')) foo(js);
+js	object_with_unique	object_without_unique	array_with_unique	array_without_unique
+{"a":1, "b":2, "c":3}	1	1	0	0
+{"a":1, "a":2, "a":3}	0	1	0	0
+[{"a":"1"},{"c":"2","c":"3"}]	0	0	0	1
+[{"a":"1"},{"b":"2","c":"3"}]	0	0	1	1
+[1,2,3]	0	0	1	1
+[1,1,1]	0	0	1	1
+[]	0	0	1	1
+123	0	0	0	0
+"string"	0	0	0	0
+true	0	0	0	0
+null	0	0	0	0
+# Test with table data
+CREATE TABLE test_json (
+json_col JSON,
+text_col TEXT
+);
+INSERT INTO test_json VALUES 
+('{"name":"Alice", "age":25}', '{"name":"Alice", "age":25}'),
+('[1,2,3]', '[1,2,3]'),
+('42', '42'),
+('"hello"', '"hello"'),
+('true', 'true'),
+('null', 'null'),
+('"invalid"', 'invalid'),
+('{"key1":1, "key2":2}', '{"key1":1, "key2":2}'),
+('{"key1":1, "key1":2}', '{"key1":1, "key1":2}');
+SELECT 
+json_col,
+json_col IS JSON as is_json,
+json_col IS JSON VALUE as is_value,
+json_col IS JSON SCALAR as is_scalar,
+json_col IS JSON OBJECT as is_object,
+json_col IS JSON ARRAY as is_array,
+json_col IS JSON OBJECT WITH UNIQUE KEYS as object_with_unique,
+json_col IS JSON OBJECT WITHOUT UNIQUE KEYS as object_without_unique
+FROM test_json;
+json_col	is_json	is_value	is_scalar	is_object	is_array	object_with_unique	object_without_unique
+{"name":"Alice", "age":25}	1	1	0	1	0	1	1
```

Comment:
Result looks incorrect. Please fix. object_with_unique and object_without_unique cant be both true at the same time.